### PR TITLE
KOGITO-7385: Implement acceptance tests for message collaboration in BPMN editor

### PR DESCRIPTION
**Pull request by domhanak**
JIRA: https://issues.redhat.com/browse/KOGITO-7385
BAPL: https://issues.redhat.com/browse/BAPL-2161

### CODE REVIEW

File: packages/vscode-extension-pack-kogito-kie-editors/it-tests/helpers/bpmn/PropertiesPanelHelper.ts [hunk 1]
```diff
@@ -188,4 +191,19 @@ export default class PropertiesPanelHelper {
...
+
+  /**
+   * Get a correlation modal helper class that allows checking and asserting
+   * Collaborations on top-level of process.
+   *
+   * @returns CorrelationModalHelper class that is initialized
+   */
+  public async getCollerationModalHelper(): Promise<CorrelationModalHelper> {
+    await this.expandPropertySection(PropertiesPanelSection.COLLABORATION);
+    const correllationsButton = this.root.findElement(By.xpath("//button[@id='correlationsButton']"));
```
Comment (tomasdavidorg) [inline packages/vscode-extension-pack-kogito-kie-editors/it-tests/helpers/bpmn/PropertiesPanelHelper.ts]:
```suggestion
    const correllationsButton = await this.root.findElement(By.xpath("//button[@id='correlationsButton']"));
```

File: packages/vscode-extension-pack-kogito-kie-editors/it-tests/helpers/bpmn/CorrelationModalHelper.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  public async assertCorrelationsContain(tested: Correlation): Promise<void> {
+    const correlations = await this.getCorrelations();
+    for (const correlation of correlations) {
+      if (
+        (tested.getId() === correlation.getId() &&
+          tested.getName() === correlation.getName() &&
+          tested.getPropertyId() === correlation.getPropertyId(),
+        tested.getPropertyName() === correlation.getPropertyName(),
+        tested.getPropertyType() === correlation.getPropertyType())
+      ) {
```
Comment (tomasdavidorg) [inline packages/vscode-extension-pack-kogito-kie-editors/it-tests/helpers/bpmn/CorrelationModalHelper.ts]: Maybe the formatting is different but the parentheses are not necessary.
```suggestion
      if (tested.getId() === correlation.getId() &&
          tested.getName() === correlation.getName() &&
          tested.getPropertyId() === correlation.getPropertyId(),
          tested.getPropertyName() === correlation.getPropertyName(),
          tested.getPropertyType() === correlation.getPropertyType()
      ) {
```

File: packages/vscode-extension-pack-kogito-kie-editors/it-tests/helpers/bpmn/CorrelationModalHelper.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import Correlation from "./Correlation";
+
+/**
+ * Class for accessing expanded BPMN Diagram explorer panel.
+ */
+export default class CorrelationModalHelper {
+  constructor(private readonly root: WebElement) {}
+
+  public async closeModal(): Promise<void> {
+    const closeBtn = this.root.findElement(By.xpath(".//button[@data-dismiss='modal']"));
```
Comment (tomasdavidorg) [inline packages/vscode-extension-pack-kogito-kie-editors/it-tests/helpers/bpmn/CorrelationModalHelper.ts]:
```suggestion
    const closeBtn = await this.root.findElement(By.xpath(".//button[@data-dismiss='modal']"));
```

File: packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts [hunk 4]
```diff
@@ -369,4 +369,55 @@ describe("Editors are loading properly", () => {
...
+    );
+    correllationsModalHelper.assertCorrelationsContain(
+      new Correlation("CK_ID_2", "CK Name 1", "CP_ID_2", "CP Name 2", "Integer")
+    );
+    correllationsModalHelper.assertCorrelationsContain(
+      new Correlation("CK_ID_3", "CK Name 2", "CP_ID_3", "CP Name 3", "Boolean")
+    );
+    correllationsModalHelper.assertCorrelationsContain(
+      new Correlation("CK_ID_4", "CK Name 3", "CP_ID_4", "CP Name 4", "java.util.ArrayList<String>")
+    );
```
Comment (tomasdavidorg) [inline packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts:393]: Would it be possible to add a size check?

File: packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts [hunk 5]
```diff
@@ -369,4 +369,55 @@ describe("Editors are loading properly", () => {
...
+
+    await correllationsModalHelper.closeModal();
+
+    let diagramExplorer = await bpmnEditorTester.openDiagramExplorer();
+    await diagramExplorer.selectDiagramNode("startMessage");
+    propertiesPanel = await bpmnEditorTester.openDiagramProperties();
+    await propertiesPanel.expandPropertySection(PropertiesPanelSection.CORRELATION);
+    await propertiesPanel.assertPropertyValue("Property", "CK Name 1 - CP Name 1", "select");
+    await propertiesPanel.assertPropertyValue("Message Expression Type", "Object", "div/select");
+    await propertiesPanel.assertPropertyValue("Data Expression Type", "Object", "div/select");
```
Comment (tomasdavidorg) [inline packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts:403]: Also here: Would it be possible to add a size check?

File: packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts [hunk 6]
```diff
@@ -369,4 +369,55 @@ describe("Editors are loading properly", () => {
...
+    await propertiesPanel.assertPropertyValue("Message Expression Type", "Integer", "div/select");
+    await propertiesPanel.assertPropertyValue("Data Expression Type", "Integer", "div/select");
+
+    diagramExplorer = await bpmnEditorTester.openDiagramExplorer();
+    await diagramExplorer.selectDiagramNode("catchMessage");
+    propertiesPanel = await bpmnEditorTester.openDiagramProperties();
+    await propertiesPanel.expandPropertySection(PropertiesPanelSection.CORRELATION);
+    await propertiesPanel.assertPropertyValue("Property", "CK Name 3 - CP Name 4", "select");
+    await propertiesPanel.assertPropertyValue("Message Expression Type", "java.util.ArrayList<String>", "div/select");
+    await propertiesPanel.assertPropertyValue("Data Expression Type", "java.util.ArrayList<String>", "div/select");
```
Comment (tomasdavidorg) [inline packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts:419]: Also here: Would it be possible to add a size check?

File: packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts [hunk 7]
```diff
@@ -369,4 +369,55 @@ describe("Editors are loading properly", () => {
...
+    await propertiesPanel.assertPropertyValue("Message Expression Type", "Object", "div/select");
+    await propertiesPanel.assertPropertyValue("Data Expression Type", "Object", "div/select");
+
+    diagramExplorer = await bpmnEditorTester.openDiagramExplorer();
+    await diagramExplorer.selectDiagramNode("boundaryEventMessage");
+    propertiesPanel = await bpmnEditorTester.openDiagramProperties();
+    await propertiesPanel.expandPropertySection(PropertiesPanelSection.CORRELATION);
+    await propertiesPanel.assertPropertyValue("Property", "CK Name 1 - CP Name 2", "select");
+    await propertiesPanel.assertPropertyValue("Message Expression Type", "Integer", "div/select");
+    await propertiesPanel.assertPropertyValue("Data Expression Type", "Integer", "div/select");
```
Comment (tomasdavidorg) [inline packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts:411]: Also here: Would it be possible to add a size check?

File: packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts (hunk 5 above)
Comment (domhanak) [reply packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts:403]: Here it does not make sense.

File: packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts (hunk 7 above)
Comment (domhanak) [reply packages/vscode-extension-pack-kogito-kie-editors/it-tests/extension-editors.test.ts:411]: Here it also does not make sense.

**Review by tomasdavidorg — APPROVED** [review]

**Review by tiagobento — APPROVED** [review]

**Pull request merged**